### ISSUE TITLE
`F841` false negative when access from inner function

### ISSUE THREAD

**Issue by nikolaik** [charliermarsh/ruff#146]
```python
# yolo.py:
def outer():
    x = 1

    def inner():
        print(x)

    inner()
```
```console
$ ruff yolo.py
yolo.py:2:5: F841 Local variable `x` is assigned to but never used

Found 1 error(s).
```

If I remove `def outer` and dedent the rest there is no lint error.

I see similar behaviour when `inner` is defined using `lambda`.
